MaxText/pyconfig.py

```diff
@@ -64,19 +65,26 @@ def validate_kv_quant_axis(s: str, quantize_kvcache: bool) -> None:
   if quantize_kvcache and s == "":
     raise ValueError("kv_quant_axis can not be '' when quantize_kvcache is True")
 
+def validate_attention_kernel(s: str) -> None:
+  valid_attention_kernels = ("autoselected", "dot_product", "flash", "cudnn_flash_te")
+  if s not in valid_attention_kernels:  # currently supported attention
+    raise ValueError("Invalid attention kernel was passed. Valid options ", valid_attention_kernels)
+
 def validate_attention_type(s: str) -> None:
-  valid_attention_types = ("autoselected", "dot_product", "flash", "cudnn_flash_te")
+  valid_attention_types = (attention_type.value for attention_type in AttentionType)
```

Comment:
nit maybe we can use strings to be similar to other styles in the repository?

Reply:
no strong preference here, what do @khatwanimohit and @gobbleturk think?

Reply:
Personally I like it this way. It looks more pythonic and we should have done this for the long list of model names as well.

Reply:
Ya this looks better to me, ideally we should refactor the code to avoid strings where possible like done here